public/components/event_analytics/explorer/sidebar/sidebar.tsx

```diff
@@ -54,6 +54,18 @@ export const Sidebar = (props: ISidebarProps) => {
   const [showFields, setShowFields] = useState<boolean>(false);
   const [searchTerm, setSearchTerm] = useState<string>('');
 
+  // list of every single field object
+  const [allFields, setAllFields] = useState<IField[]>([]);
+
+  // method to return the type of a field from its name
+  const getFieldTypes = (newFieldName: string) => {
+    let fieldType: string = '';
+    allFields.map((field) => {
```

Comment:
This is a quick fix. Ideally, we can do away with `allFields` and just loop over `explorerFields.availableFields` and `explorerFields.selectedFields` to get type.

Reply:
Ok I can make the change